database/src/main/java/com/firebase/ui/database/ObservableSnapshotArray.java

```diff
@@ -59,20 +61,27 @@ public ChangeEventListener addChangeEventListener(@NonNull ChangeEventListener l
             listener.onDataChanged();
         }
 
+        if (!wasListening) { onCreate(); }
+
         return listener;
     }
 
+    @CallSuper
```

Comment:
We should add JavaDoc on `onCreate` and `onDestroy` to describe what they do and when a subclass should override them.